src/test/java/org/jabref/logic/importer/fetcher/AstrophysicsDataSystemTest.java

```diff
@@ -198,4 +200,18 @@ public void testPerformSearchByLuceyPaulEntry() throws Exception {
         Optional<BibEntry> fetchedEntry = fetcher.performSearchById("2000JGR...10520297L");
         assertEquals(Optional.of(luceyPaulEntry), fetchedEntry);
     }
+
+    @Test
+    public void performSearchByQueryPaged_searchLimitsSize() throws Exception {
+        Page<BibEntry> page = fetcher.performSearchPaged("author:\"A\"", 0);
+        assertEquals(fetcher.getPageSize(), page.getSize(), "fetcher return wrong page size");
+    }
+
+    @Test
+    public void performSearchByQueryPaged_twoPagesNotEqual() throws Exception {
+        Page<BibEntry> page = fetcher.performSearchPaged("author:\"A\"", 0);
+        Page<BibEntry> page2 = fetcher.performSearchPaged("author:\"A\"", 1);
```

Comment:
I think, in the UI it will be hard to always pass the query. Why not adding an intermediate layer `searchPagesFactory = fetcher.searchPagesFactory("author:...")` and then `searchPagesFactory.getPage(0)`?

Reply:
The stateless argument applies here too. I would prefer the viewmodel to handle state